File: GoGolang/three.go
package main
import "fmt"
func main(){
	var n int
	fmt.Print("Введите количество чисел: ")
	fmt.Scan(&n)
	for i := 0; i < n; i++ {
		var num int
		fmt.Print("Введите число: ")
		fmt.Scan(&num)
		reversed := rev(num)
		fmt.Println(reversed)
	}
}
func rev(x int) int{
	neg := x < 0
	rev := 0
	if neg {
		x = -x
	}
	for x > 0 {
		rev = rev * 10 + x % 10
		x = x / 10
	}
	if neg {
		rev = -rev
	}
	return rev
}

File: php/three.php
<?php
function rev($x){
    $neg = $x < 0;
    $rev = 0;
    $x = abs($x);
    while($x > 0){
        $rev = $rev * 10 + $x % 10;
        $x = intdiv($x, 10);
    }
    return $neg ? -$rev : $rev;
}
echo 'Введите количество чисел: ';
$n = (int)trim(fgets(STDIN));
for ($i=0; $i<$n; $i++){
    echo 'Введите число: ';
    $num = (int)trim(fgets(STDIN));
    $reversed = rev($num);
    echo $reversed . PHP_EOL;
}
